services/static-webserver/client/source/class/osparc/auth/ui/RequestAccount.js

```diff
@@ -88,10 +88,25 @@ qx.Class.define("osparc.auth.ui.RequestAccount", {
       });
       this._form.add(postalCode, this.tr("Postal code"), null, "postalCode");
 
-      const country = new qx.ui.form.TextField().set({
+      const country = new qx.ui.form.SelectBox().set({
         required: true
       });
       doubleSpaced.push(country);
+      const countries = osparc.store.StaticInfo.getInstance().getCountries();
+      countries.forEach(c => {
+        const cItem = new qx.ui.form.ListItem(c.name, null, c.alpha2).set({
+          rich: true
+        });
+        country.add(cItem);
+      })
+      fetch("https://ipapi.co/json")
```

Comment:
Sugestion: if it fails the request I would retry it 3 times then give up, since you are relying on an external service.

Q: what happens if this fails, will I not get a country automatically selected? In that case I'd say it's fine

Reply:
Correct.